### PR TITLE
Scan directly with twistcli when checkov can't like gradle files

### PR DESCRIPTION
**Pull request by matansha**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

checkov have problem to scan some files, so in that cases we will try to scan with twistcli on the code repo itself.

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/sca_package/scanner.py [hunk 1]
```diff
@@ -53,6 +57,28 @@ async def run_scan_multi(
...
+                # it avoids us from crashing, which happens when using multiprocessing via Pycharm's debug-mode
+                logging.warning("Running the scans in sequence for avoiding crashing when running via Pycharm")
+                scan_twist_results = []
+                for idx, input_path in enumerate(input_paths):
+                    if scan_results[idx]["vulnerabilities"] is None:
+                        scan_twist_results.append(await self.execute_twistcli_scan(input_path))
+            else:
+                scan_twist_results = await asyncio.gather(*[
+                    self.execute_twistcli_scan(input_path) for idx, input_path in enumerate(input_paths) if scan_results[idx]["vulnerabilities"] is None
+                ])
```
Comment (nimrodkor) [inline checkov/sca_package/scanner.py:74]: How about this, and then no need for the re-fill?
```suggestion
                scan_twist_results = await asyncio.gather(*[
                    self.execute_twistcli_scan(input_path) if scan_results[idx]["vulnerabilities"] is None else scan_results[idx] for idx, input_path in enumerate(input_paths) 
                ])
```

**Review by nimrodkor — COMMENTED** [review]
Cool @matansha !

File: checkov/sca_package/scanner.py [hunk 2]
```diff
@@ -111,3 +137,46 @@ def parse_api_result(self, origin_file_path: Path, response: str) -> dict[str, A
...
+                    try:
+                        os.makedirs(bridgecrew_dir)
+                    except FileExistsError:
+                        # In multi-processing, this might meet a race condition
+                        pass
+                image_scanner.cleanup_scan()
+                docker_image_scanning_integration.download_twistcli(image_scanner.twistcli_path)
+        except Exception:
+            logging.error("Failed to setup docker image scanning", exc_info=True)
+            raise
```
[the remark this answers is not in the document]
Comment (matansha) [reply checkov/sca_package/scanner.py]: done

**Review by nimrodkor — APPROVED** [review]

**Review by gruebel — APPROVED** [review]
noice 🍻

**Pull request merged**